File: frontend/mobile/src/6_entities/child/model/child.ts
import {SchoolClass} from '../../school-class/model/school-class';
import {MealPlan} from './meal-plan';

export type Child = {
    id: string,
    lastName: string,
    firstName: string,
    schoolClass: SchoolClass,
    mealPlan: MealPlan
}

File: backend/app/children/domain/child.py
from pydantic.dataclasses import dataclass

from app.children.domain.meal_plan import MealPlan
from app.children.domain.school_class import SchoolClass
from app.shared.domain import Entity, ValueObject


@dataclass(eq=True, frozen=True)
class ChildID(ValueObject):
    value: str


@dataclass(eq=True, frozen=True)
class LastName(ValueObject):
    value: str


@dataclass(eq=True, frozen=True)
class FirstName(ValueObject):
    value: str


@dataclass
class Child(Entity):
    id: ChildID
    last_name: LastName
    first_name: FirstName
    school_class: SchoolClass
    meal_plan: MealPlan
